File: STP/src/main/java/com/summit/stp/order/domain/model/OrderStatus.java
package com.summit.stp.order.domain.model;

import lombok.Getter;
import com.summit.stp.shared.exception.ParameterException;

/**
 * 订单状态值对象 (枚举)
 */


@Getter
public enum OrderStatus {
    PENDING(0, "待支付"),
    PAID(1, "已支付"),
    COMPLETED(2, "已完成"),
    CANCELLED(3, "已取消");

    private final int code;
    private final String description;

    OrderStatus(int code, String description) {
        this.code = code;
        this.description = description;
    }

    public static OrderStatus fromCode(Integer code) {
        if (code == null) {
            return PENDING;
        }
        for (OrderStatus status : values()) {
            if (status.code == code) {
                return status;
            }
        }
        throw new ParameterException("未知的订单状态码: " + code);
    }
}

File: frontend-client/src/views/payment/types/orderStatus.ts
export enum OrderStatus {
  PENDING = 0,
  PAID = 1,
  COMPLETED = 2,
  CANCELLED = 3,
}

export namespace OrderStatus {
  export interface Meta {
    code: OrderStatus
    description: string
  }

  const list: Meta[] = [
    { code: OrderStatus.PENDING, description: '待支付' },
    { code: OrderStatus.PAID, description: '已支付' },
    { code: OrderStatus.COMPLETED, description: '已完成' },
    { code: OrderStatus.CANCELLED, description: '已取消' },
  ]

  export function getList(): Meta[] {
    return list
  }

  export function getDescription(code: OrderStatus | number | string): string {
    const found = list.find((item) => item.code === fromCode(code))
    return found ? found.description : '未知状态'
  }

  export function fromCode(code: number | string | undefined | null): OrderStatus | undefined {
    if (code === undefined || code === null) return undefined
    if (typeof code === 'string') {
      const upper = code.toUpperCase()
      if (upper === 'PENDING') return OrderStatus.PENDING
      if (upper === 'PAID') return OrderStatus.PAID
      if (upper === 'COMPLETED') return OrderStatus.COMPLETED
      if (upper === 'CANCELLED') return OrderStatus.CANCELLED
    }
    const num = typeof code === 'number' ? code : parseInt(String(code), 10)
    if (isNaN(num)) return undefined
    const found = list.find((item) => item.code === num)
    return found ? found.code : undefined
  }
}
